### PR TITLE
Update documentation for ExternalName in 1.21

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

File: content/en/docs/ops/configuration/traffic-management/traffic-routing/index.md [hunk 1]
```diff
@@ -16,9 +16,28 @@ To understand what is happening in your mesh, it is important to understand how
...
+  This is necessary to identify which backend to route traffic to, and which policies to apply.
+  For example, we may read the `Host` header of `http.ns.svc.cluster.local` and identify the request is intended for the `http` Service.
+  More information on how this matching works can be found below.
+* The "backend" refers to where we send traffic once we have matched it.
+  Using the example above, after identifying the request as targeting the `http` Service, we would send it to an endpoint in that Service.
+  However, this selection is not always so simple; Istio allows customization of this logic, through `VirtualService` routing rules.
+
+Standard Kubernetes networking has these same concepts, too, but they are much simpler and generally hidden.
+When a `Service` is created, there is typically an associated frontend -- the automatically created DNS name (such as `http.ns.svc.cluster.local`),
+and an automatically create IP address to represent the service (the `ClusterIP`).
```
Comment (kfaseela) [inline content/en/docs/ops/configuration/traffic-management/traffic-routing/index.md:33]:
```suggestion
and an automatically created IP address to represent the service (the `ClusterIP`).
```

File: content/en/docs/ops/configuration/traffic-management/traffic-routing/index.md [hunk 2]
```diff
@@ -16,9 +16,28 @@ To understand what is happening in your mesh, it is important to understand how
...
+Similarly, a backend is also create - the `Endpoints` or `EndpointSlice` - which represents all of the `Pod`s selected by the service.
```
Comment (kfaseela) [inline content/en/docs/ops/configuration/traffic-management/traffic-routing/index.md:34]:
```suggestion
Similarly, a backend is also created - the `Endpoints` or `EndpointSlice` - which represents all of the `Pod`s selected by the service.
```

File: content/en/docs/ops/configuration/traffic-management/traffic-routing/index.md [hunk 3]
```diff
@@ -16,9 +16,28 @@ To understand what is happening in your mesh, it is important to understand how
...
+  For example, we may read the `Host` header of `http.ns.svc.cluster.local` and identify the request is intended for the `http` Service.
+  More information on how this matching works can be found below.
+* The "backend" refers to where we send traffic once we have matched it.
+  Using the example above, after identifying the request as targeting the `http` Service, we would send it to an endpoint in that Service.
+  However, this selection is not always so simple; Istio allows customization of this logic, through `VirtualService` routing rules.
+
+Standard Kubernetes networking has these same concepts, too, but they are much simpler and generally hidden.
+When a `Service` is created, there is typically an associated frontend -- the automatically created DNS name (such as `http.ns.svc.cluster.local`),
+and an automatically created IP address to represent the service (the `ClusterIP`).
+Similarly, a backend is also created - the `Endpoints` or `EndpointSlice` - which represents all of the `Pod`s selected by the service.
```
Comment (kfaseela) [inline content/en/docs/ops/configuration/traffic-management/traffic-routing/index.md:34]:
```suggestion
Similarly, a backend is also created - the `Endpoints` or `EndpointSlice` - which represents all of the pods selected by the service.
```

File: content/en/docs/ops/configuration/traffic-management/traffic-routing/index.md [hunk 4]
```diff
@@ -16,9 +16,28 @@ To understand what is happening in your mesh, it is important to understand how
 This document describes low level implementation details. For a higher level overview, check out the traffic management [Concepts](/docs/concepts/traffic-management/) or [Tasks](/docs/tasks/traffic-management/).
 {{< /warning >}}
 
+## Frontends and backends
+
+In traffic routing in Istio, there are two primary phases:
+
+* The "fronted" refers to how we match the type of traffic we are handling.
```
Comment (ilrudie) [inline content/en/docs/ops/configuration/traffic-management/traffic-routing/index.md:23]:
```suggestion
* The "frontend" refers to how we match the type of traffic we are handling.
```

File: content/en/docs/ops/configuration/traffic-management/traffic-routing/index.md [hunk 5]
```diff
@@ -95,15 +115,33 @@ However, with Istio the port must be declared in the Service, or it will [not be
...
+If this is a service Istio knows about, it will be routed as described [above](#tcp).
+If not, it will be handled as [unmatched traffic](#unmatched-traffic)
+
+For HTTP and TLS, which match on hostname, things are a bit different.
+If the target service (`concrete.example.com`) is a service Istio knows about, then the alias hostname (`alias.default.svc.cluster.local`) will be added
+as an _additional_ match to the [TLS](#tls) or [HTTP](#http) matching.
+If not, there will be no changes, so it will be handled as [unmatched traffic](#unmatched-traffic).
+
+An `ExternalName` service can never be a [backend](#frontends-and-backends) on its own.
+Instead, it is only ever adding additional [frontend](#frontends-and-backends) matches to existing Services.
```
Comment (ilrudie) [inline content/en/docs/ops/configuration/traffic-management/traffic-routing/index.md:141]: "adding additional" seems weird to me, YMMV
```suggestion
Instead, it is only ever used as additional [frontend](#frontends-and-backends) matches to existing Services.
```

**Review by ilrudie — COMMENTED** [review]
LGTM, 1 typo and 1 nitty suggestion about wording

**Review by kfaseela — APPROVED** [review]

**Pull request merged**